assets/src/edit-story/components/library/panes/shared/chipGroup/index.js

```diff
@@ -0,0 +1,211 @@
+const ExpandButton = styled(Button).attrs({
+  type: BUTTON_TYPES.TERTIARY,
+  size: BUTTON_SIZES.SMALL,
+  variant: BUTTON_VARIANTS.CIRCLE,
+})`
+  display: flex;
+  position: absolute;
+  bottom: -24px;
+  background: ${({ theme }) => theme.colors.bg.tertiary};
+  max-height: none;
+  width: 32px;
+  height: 32px;
+  ${({ isExpanded }) => isExpanded && 'transform: matrix(1, 0, 0, -1, 0, 0);'}
+  visibility: inherit;
+  align-self: center;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ChipGroup = ({ items, selectedItemId, selectItem, deselectItem }) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const sectionRef = useRef();
+  const innerContainerRef = useRef();
+
+  const itemRefs = useRef([]);
+
+  const [focusedRowOffset, setFocusedRowOffset] = useState(0);
+
+  const handleClick = useCallback(
+    (selected, id) => {
+      if (selected && id !== null) {
+        deselectItem();
+      } else {
+        setIsExpanded(false);
+        selectItem(id);
+      }
+    },
+    [deselectItem, selectItem]
+  );
+
+  const handleExpandClick = useCallback(
+    () => setIsExpanded((currentExpanded) => !currentExpanded),
+    []
+  );
+
+  useExpandAnimation({
+    sectionRef,
+    innerContainerRef,
+    isExpanded,
+    setFocusedRowOffset,
+  });
+
+  useHandleRowVisibility({
+    isExpanded,
+    innerContainerRef,
+    selectedItemId,
+    setFocusedRowOffset,
+    itemRefs,
+  });
+
+  const hasItems = items.length > 0;
+  useRovingTabIndex({ ref: sectionRef }, [isExpanded]);
+  useKeyDownEffect(
+    sectionRef,
+    !isExpanded ? 'down' : '',
+    () => hasItems && setIsExpanded(true),
+    [isExpanded, hasItems]
+  );
+
+  const containerId = useMemo(() => `pill-group-${uuidv4()}`, []);
+  return (
+    <Section ref={sectionRef}>
+      {hasItems && (
+        <>
+          <Container
+            id={containerId}
+            isExpanded={isExpanded}
+            role="listbox"
+            aria-label={__('List of filtering options', 'web-stories')}
+          >
+            <InnerContainer
+              role="presentation"
+              ref={innerContainerRef}
+              style={{ transform: `translateY(-${focusedRowOffset}px` }}
+            >
+              {items.map((item, i) => {
+                const { id, label } = item;
+                const selected = id === selectedItemId;
+
+                return (
+                  <Chip
+                    key={id}
+                    role="option"
+                    ref={(el) => {
+                      itemRefs.current[id] = el;
+                    }}
+                    active={selected}
+                    aria-selected={selected}
+                    onClick={() => handleClick(selected, id)}
+                    // The first or selected category will be in focus for roving
+                    // (arrow-based) navigation initially.
+                    tabIndex={i === 0 || selected ? 0 : -1}
```

Comment:
It looks like currently, there can be two chips that can be tabbed to -- the selected one and the chip that was last focused on while using the `useRovingTabIndex`. 
To test out:
1. Tab into the chip group and move the focus with arrow keys without selecting any new chip.
2. Now tab through the chip group -- both the selected chip and the last chip that had focus via using the arrow keys are tabbable now:
![chipgroup](https://user-images.githubusercontent.com/3294597/113390836-ca17d580-9392-11eb-99f1-e79ff4e08b93.gif)

Reply:
Maybe that's expected? Not sure :)

Reply:
This should not happen, no.

Reply:
> This should not happen, no.

Sure can take this out. It was taken from the `RadioGroup` that was used here before